cmd/medius/images/push.go

```diff
@@ -247,6 +238,74 @@ func (b *buildAndPublish) readArtifact(artifactReader http.ReadCloserWithChecksu
 	return file.Name(), nil
 }
 
+func (b *buildAndPublish) buildImages(entry *common.Entry) ([]v1.Image, []string, error) {
+	var images []v1.Image
+	var artifacts []string
+
+	rebuildNeeded, err := b.rebuildNeeded(entry)
+	if err != nil {
+		return nil, nil, err
+	}
+	if !rebuildNeeded && !b.Options.PublishImagesOptions.ForceBuild {
+		b.Log.Info("Nothing to do.")
+		return nil, nil, nil
+	}
+
+	for i := range entry.Artifacts {
+		metadata := entry.Artifacts[i].Metadata()
+		artifactInfo, err := entry.Artifacts[i].Inspect()
+		if err != nil {
+			return nil, nil, fmt.Errorf("error introspecting artifact %q: %v", metadata.Describe(), err)
+		}
+
+		b.Log.Infof("Rebuild needed, downloading %q ...", artifactInfo.DownloadURL)
+		file, err := b.getArtifact(artifactInfo)
+		if err != nil {
+			return nil, nil, err
+		}
+		artifacts = append(artifacts, file)
+
+		b.Log.Info("Building containerdisk ...")
+		image, err := build.ContainerDisk(file,
+			artifactInfo.ImageArchitecture,
+			build.ContainerDiskConfig(artifactInfo.SHA256Sum, metadata.AdditionalLabels))
+		if err != nil {
+			return nil, nil, fmt.Errorf("error creating the containerdisk : %v", err)
+		}
+		if errors.Is(b.Ctx.Err(), context.Canceled) {
+			return nil, nil, b.Ctx.Err()
+		}
+		images = append(images, image)
+	}
+
+	return images, artifacts, nil
+}
+
+func (b *buildAndPublish) rebuildNeeded(entry *common.Entry) (bool, error) {
+	if len(entry.Artifacts) == 0 {
+		err := errors.New("entry has no artifacts to check for rebuild")
+		b.Log.Error(err)
+		return false, err
+	}
+
+	for i := range entry.Artifacts {
```

Comment:
use `for _, artifact := range entry.Artifacts {`
The code will be better readable

Reply:
Thanks. I'd prefer to leave it as `entry.Artifacts[i]`.